server/service/src/sync/translations/master_list.rs

```diff
@@ -47,6 +49,35 @@ impl SyncTranslation for MasterListTranslation {
         };
         Ok(PullTranslateResult::upsert(result))
     }
+
+    // Soft deletes were implemented in OG months after program requisitions was
+    // rolled out, so previously hard deleted records may be gone even if they
+    // are linked to program. Recreate these records to keep the FK link.
+    fn try_translate_from_delete_sync_record(
+        &self,
+        connection: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<PullTranslateResult, anyhow::Error> {
+        let program =
+            ProgramRowRepository::new(connection).find_one_by_id(&sync_record.record_id)?;
+        let master_list =
+            MasterListRowRepository::new(connection).find_one_by_id(&sync_record.record_id)?;
+
+        if let (Some(_), Some(master_list)) = (program, master_list) {
+            let result = MasterListRow {
+                id: master_list.id.clone(),
+                name: master_list.name.clone(),
+                code: master_list.code.clone(),
+                description: master_list.description.clone(),
+                is_active: false,
```

Comment:
Nit: don't need to clone here and could use `..master_list` syntax to take all existing fields...